errtracker/errtracker.go

```diff
@@ -79,6 +81,20 @@ func readMachineID() ([]byte, error) {
 	return nil, fmt.Errorf("cannot report: no suitable machine id file found")
 }
 
+// detectPartialDpkgUpdate returns true if a partial update of the snapd dpkg
+// package is detected. This is done by inspecting the apparmor profile of
+// snap-confine which is classified as a conffile and this has different update
+// semantics than that of regular files.
+//
+// This probe is here to aid in resolving the following issue:
+// https://forum.snapcraft.io/t/test-failures-with-cannot-create-lock-directory-run-snapd-lock/390/
+func detectPartialDpkgUpdate() string {
```

Comment:
AFAIR this is also happening when a partial update (broken mid way, before snapd is configured) happens so this is why I called it like that. I can rename it to something neutral like `detectSnapConfineDpkgNew`